### DISCUSSION TITLE
Information viz for lineages to unshown nodes in draw_node

### DISCUSSION THREAD

**Discussion by hyanwong** [kitchensjn/tskit_arg_visualizer#113] [General]
This is a bit trivial, but I've noticed that having a single line upwards or downwards from a node in the `draw_node()` view might be considered misleading if it leads to more than 1 child or parent. This is because if could imply that all the children are "grouped together" in some way, rather than representing a large polytomy

<img width="303" alt="Screenshot 2024-12-13 at 11 32 21" src="https://github.com/user-attachments/assets/680c2d61-4066-4b77-ba90-1deac8582dd0" />

I recently solved this in tskit by taking advantage of the fact that it uses an "ortho"-type view, so we can simply draw a dotted horizontal line (sadly I don't think this would work in the draw_node viz here):
<img width="823" alt="Screenshot 2024-12-13 at 11 38 09" src="https://github.com/user-attachments/assets/71cdb2ab-21f0-4f58-9cdd-1d6a5bc1a2ef" />

The way we solved it when previously showing subgraphs was to draw multiple dotted lines, which works fine for small polytomies, but risks weird behaviour e.g. for polytomies of 100 nodes or more:

<img width="230" alt="Screenshot 2024-12-13 at 11 42 35" src="https://github.com/user-attachments/assets/c058c4fc-69a6-4278-a1a5-5ecc40d5e644" />

I'm not sure there is a good solution here, but perhaps you have interesting suggestions @kitchensjn ?

**Comment by hyanwong** [comment]
Note also that in the tskit tree viz, I display both the number of unshown lineages descending from the polytomy (in bold), *and* the total number of descendant samples. However, finding the number of descendant samples in an ARG is not a trivial matter (a weighted version would probably involve something like https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TreeSequence.mean_descendants, which is not instantaneous: I just tried it on a ts with 11435813 nodes  and 65790397 edges and it took ~ 45 secs), so that might not be a good idea for an entire ARG? On the other hand, it's relatively quick when there are fewer edges: `mean_descendants()` seemed instant for 303735 nodes, 303861 edges (I suspect it is linear in the number of edges), so we could pre-calculate that and add it to the internal tskit_visualizer node table.

**Comment by kitchensjn** [comment]
I missed this over the holidays and may need to check that new discussion threads give me notifications. I can see your point. Although to confirm, the number associated with the dashed line refers to the number of parents (or children) that of that specific node that are note shown in the subgraph. So if that node is a polytomy with 10 children and only one is in the subgraph, the there will be a 9 underneath the dashed line. Let me know if that needs to be clarified better (I actually don't think there's any documentation on the specifics of `draw_node()` so I'll add that to the to do list). If the triangle symbol that you use in "pack_untracked_polytomies" plot would make it more clear when there are multiple child lines, I could look into adding that. I'm hesitant to add many more lines coming off the as that could get really cluttered.

Right now, there isn't an indication of how many leaves are found beneath a node and that would be tricky to calculate from the `D3ARG` object; it would be best to do that in `tskit` before converting over, as you've described. But I'm not sure then how you would update that number when you draw different subgraphs. Sometimes the leaf is included in the subgraph, so you wouldn't want to double count, but other times not.
